### ISSUE TITLE
PNG - resize and then save as JPG - extremely poor quality

### ISSUE THREAD

**Issue by PhyxionNL** [SixLabors/ImageSharp#2841]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of ImageSharp
- [X] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [X] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### ImageSharp version

3.1.6

### Other ImageSharp packages and versions

3.1.6

### Environment (Operating system, version and so on)

Windows

### .NET Framework version

9.0

### Description

Source:
![PNG](https://github.com/user-attachments/assets/50bb9176-a0f0-4c6e-a70b-0ebfa90b0e5f)

Destination:
![JPG](https://github.com/user-attachments/assets/09c1606b-e87d-475b-99da-3dddf867d92c)

When resizing with say Paint.NET:
![JPG alt](https://github.com/user-attachments/assets/9feb76f9-bf4b-47cf-a2e7-397f6ef5e8c9)

### Steps to Reproduce

```
  using (var img = Image.Load(@"example.png"))
  {
      img.Mutate(x => x.Resize(64, 64));

      img.Save(@"out.jpg", new JpegEncoder
      {
          Quality = 100
      });
  }
```

If you save it as PNG it's OK. WebP also.

**Comment by PhyxionNL** [comment]
Just noticed that without resizing the result is very bad as well.

**Comment by JimBobSquarePants** [comment]
I guarantee that the transparent areas in the image are not really transparent.

**Comment by JimBobSquarePants** [comment]
> > I guarantee that the transparent areas in the image are not really transparent.
> 
> They absolutely are.

Sorry, what I mean is that following the resize event the "transparent" pixels surrounding each node are not true transparent e.g. (0,0,0,0). 

What Paint.NET is doing here is painting the result of the resize onto a white background so the final result is the composition between the two. 

You can replicate that behavior with the following code.

```c#
using (var img = Image.Load<Rgba32>(@"C:\Users\james\Downloads\2841.png"))
{
    img.Mutate(x => x.Resize(64, 64));

    using var bg = new Image<Rgba32>(64, 64, Color.White);

    bg.Mutate(x => x.DrawImage(img, 1F));

    bg.Save(@"C:\Users\james\Downloads\2841-out.jpg", new JpegEncoder
    {
        Quality = 100
    });
}

```
Result.

![2841-out](https://github.com/user-attachments/assets/5c578201-68a6-4a4e-b010-31c7027234ce)

If you use `Color.Transparent` (255,255,255,0) 

![2841-out - Copy](https://github.com/user-attachments/assets/e75495e8-cbcd-4892-a36c-b768d887a25e)

Technically ImageSharp is doing the correct thing here by dropping the alpha channel on save (32bit => 24bit), it just doesn't match some user expectations because graphics tools like Pain.NET tend to perform additional operations to achieve pleasing results. 

As a side note (and this does not affect output) your input image uses both (0,0,0,0) and (255,255,255,0) to represent transparency inside and outside the square.

**Comment by PhyxionNL** [comment]
![image](https://github.com/user-attachments/assets/17629f89-ba46-4884-b8f9-6911a5d6a07b)

The first is resized and saved as PNG with ImageSharp, the second is manually colored in all semi-transparent areas (I've double checked that everything surrounding has 0 transparency), and the bottom one is the JPEG.

Your workaround seems to improve things greatly, but some of the pixels still have quite some color, shouldn't they then also show up in the PNG as semi-transparent?

![image](https://github.com/user-attachments/assets/7ddfe219-68c5-4e5f-843d-7f34c1c3967f)

The selected pixel is DCFFFF for example.

**Comment by JimBobSquarePants** [comment]
The PNG output actually contains RGB data there though the A component is 0. 

There'd be subtle differences in the color components between PNG and JPEG. They're identical until encoding where the jpeg output is lossy (100 is still lossy). 

In this instance the bleed is compounded by the default chroma subsampling which blends neighboring pixels.

![image](https://github.com/user-attachments/assets/74cca338-db79-4630-947f-e148f655b78e)

Changing the subsampling to the following.

```c#
img2.Save(@"C:\Users\james\Downloads\2841-out.jpg", new JpegEncoder
{
    Quality = 100,
    ColorType = JpegEncodingColor.YCbCrRatio444
});
```
We see a marked improvement.

![image](https://github.com/user-attachments/assets/4aee27b5-c8ad-4f91-8e96-9558d28a9777)

**Comment by PhyxionNL** [comment]
> The PNG output actually contains RGB data there though the A component is 0.
> 
> There'd be subtle differences in the color components between PNG and JPEG. They're identical until encoding where the jpeg output is lossy (100 is still lossy).
> 
> In this instance the bleed is compounded by the default chroma subsampling which blends neighboring pixels.
> 

I see, wouldn't it be more logical then that JPEG rounds marginal alpha down (so it'd be similar to the actual transparent areas)?

**Comment by JimBobSquarePants** [comment]
> > The PNG output actually contains RGB data there though the A component is 0.
> > There'd be subtle differences in the color components between PNG and JPEG. They're identical until encoding where the jpeg output is lossy (100 is still lossy).
> > In this instance the bleed is compounded by the default chroma subsampling which blends neighboring pixels.
> 
> I see, wouldn't it be more logical then that JPEG rounds marginal alpha down (so it'd be similar to the actual transparent areas)?

No. As soon as we introduce unknown behavior, we break something for someone who needs it to be imperative.
